pandas/tests/frame/test_to_csv.py

```diff
@@ -790,6 +790,18 @@ def test_to_csv_unicode_index_col(self):
         df2 = read_csv(buf, index_col=0, encoding='UTF-8')
         assert_frame_equal(df, df2)
 
+    def test_to_csv_bytes(self):
+        # GH 9712
+        df = DataFrame({b'hello': ['a', b'b', u'c']})
```

Comment:
needs tests with dates and nans

these generate output characters that could be an issue here.